Inc/HALALMock/Services/SharedMemory/SharedMemory.hpp

```diff
@@ -10,8 +10,16 @@ class SharedMemory {
 
     // do not use before SharedMemory::start, uninitialized!!!
     static EmulatedPin *gpio_memory;
+	static void* emulated_state_machine{};
     static EmulatedPin &get_pin(Pin pin);
     static void start();
+	static void start_emulated_state_machine();
+	constexpr static size_t emulated_state_machine_size=16;
+	static std::string generate_shared_memory_name();
+	static std::string emulated_state_machine_name{};
+	static uint8_t* state_machine_sm{};
+	static void update_state_machine_counter(uint8_t counter);
+	static void update_current_state(uint8_t index, uint8_t state);
```

Comment:
I believe all the changes inside SharedMemory.hpp and SharedMemory.cpp should actually be in StateMachine.hpp and StateMachine.cpp. And be called on the constructor of StateMachine

Reply:
(for @g0nz4I0) Notice that the shared memory is for all state machines, not just one, so having its creation, manipulation and destruction tied to those of the state machine won't work as the shared memory will probably live even longer than the program itself. Having it here makes more sense to me as this is the class for all Shared Memory configuration and manipulation, because in that case we would have moved all the GPIO SHM stuff to `PinModel.hpp` and `PinModel.cpp` instead of creating a dedicated class.

Reply:
absolutely